lcls_tools/data_analysis/archiver.py

```diff
@@ -18,82 +20,97 @@
 
     def getDataWithTimeInterval(self, pvList, startTime, endTime, timeDelta):
-        # type: (List[str], datetime, datetime, timedelta) -> Dict[str, Dict[str, List[Union[datetime, str]]]]
+        # type: (List[str], datetime, datetime, timedelta) -> ArchiverData
         currTime = startTime
-        results = {}
+        times = []
+        values = {}
         for pv in pvList:
-            results[pv] = {"times": [], "values": []}
+            values[pv] = []
 
         while currTime < endTime:
             result = self.getDataAtTime(pvList, currTime)
-            for key, val in result.items():
-                results[key]["times"].append(currTime)
-                results[key]["values"].append(val)
+            times.append(currTime)
+            for key, val in result.values.items():
+                values[key].append(val.pop())
             currTime += timeDelta
 
-        return results
+        return ArchiverData(timeStamps=times, values=values)
 
     # returns timestamps in UTC
     def getValuesOverTimeRange(self, pvList, startTime, endTime, timeDelta=None):
-        # type: (List[str], datetime, datetime, timedelta) -> Dict[str, Dict[str, List[Union[datetime, str]]]]
+        # type: (List[str], datetime, datetime, timedelta) -> ArchiverData
 
         if timeDelta:
             return self.getDataWithTimeInterval(pvList, startTime, endTime, timeDelta)
         else:
             url = self.url_formatter.format(SUFFIX=RANGE_RESULT_SUFFIX)
 
-            results = {}
+            times = {}
+            values = {}
 
             # TODO figure out how to send all PVs at once
             for pv in pvList:
+                times[pv] = []
+                values[pv] = []
 
                 response = requests.get(url=url, timeout=TIMEOUT,
-                                        params={"pv": pv,
+                                        params={"pv"  : pv,
                                                 "from": startTime.isoformat() + "-07:00",
-                                                "to": endTime.isoformat() + "-07:00"})
+                                                "to"  : endTime.isoformat() + "-07:00"})
 
                 try:
                     jsonData = json.loads(response.text)
                     # It returns a list of len 1 for some godforsaken reason...
                     element = jsonData.pop()
-                    result = {"times": [], "values": []}
                     for datum in element[u'data']:
-                        result["times"].append(datum[u'secs'])
-                        result["values"].append(datum[u'val'])
-
-                    results[pv] = result
+                        # TODO implement filtering by BSA PV
+                        times[pv].append(datetime.fromtimestamp(datum[u'secs'])
+                                         + timedelta(microseconds=datum[u'nanos'] / 1000))
+                        values[pv].append(datum[u'val'])
 
```

Comment:
Are we confident these keys will stay the same? I'm unfamiliar. Can we pull them into a variable that would update if the key changed?

Reply:
Honestly I have no idea I'm just following the [documentation](https://slacmshankar.github.io/epicsarchiver_docs/userguide.html) I found online, but I can at least link it in some comments